data-prepper-plugins/event-json-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/event_json/EventJsonInputCodec.java

```diff
@@ -50,19 +55,20 @@ public void parse(InputStream inputStream, Consumer<Record<Event>> eventConsumer
         final JsonParser jsonParser = jsonFactory.createParser(inputStream);
 
         while (!jsonParser.isClosed() && jsonParser.nextToken() != JsonToken.END_OBJECT) {
-            if (jsonParser.getCurrentToken() == JsonToken.START_ARRAY) {
-                parseRecordsArray(jsonParser, eventConsumer);
-            }
-        }
-    }
-
-    private void parseRecordsArray(final JsonParser jsonParser, final Consumer<Record<Event>> eventConsumer) throws IOException {
-        while (jsonParser.nextToken() != JsonToken.END_ARRAY) {
-            final Map<String, Object> innerJson = objectMapper.readValue(jsonParser, Map.class);
-
-            final Record<Event> record = createRecord(innerJson);
-            if (record != null) {
-                eventConsumer.accept(record);
+            if (jsonParser.getCurrentToken() == JsonToken.START_OBJECT) {
+                final Map<String, Object> innerJson = objectMapper.readValue(jsonParser, Map.class);
+                final List<Map<String, Object>> maps = (List<Map<String, Object>>)innerJson.get(EventJsonDefines.EVENTS);
+                final String version = (String)innerJson.get(EventJsonDefines.VERSION);
```

Comment:
It may be good to move this check into a small method to help make the code more readable.